File: ios/doordeck-sdk/RNDoordeckSdk.m
#if __has_include(<React/RCTBridgeModule.h>)
#import <React/RCTBridgeModule.h>
#else
#import "RCTBridgeModule.h"
#endif


@interface RCT_EXTERN_MODULE(RNDoordeckSdk, NSObject)
+ (BOOL)requiresMainQueueSetup
{
    return YES;
}
RCT_EXTERN_METHOD(initDoordeck:(NSString *)auth darkMode:(BOOL)darkMode closeButton:(BOOL)closeButton)
RCT_EXTERN_METHOD(showUnlock:(BOOL)isNfc)
@end

File: ios/doordeck-sdk/RNDoordeckSdk.swift
import Foundation

@objc(RNDoordeckSdk)
class RNDoordeckSdk: NSObject {
  
  private var doordeck: Doordeck! = nil
  
  @objc func initDoordeck(_ auth: String, darkMode: Bool = true, closeButton: Bool = false) {
    let authToken = AuthTokenClass(auth)
    doordeck = Doordeck(authToken, darkMode: darkMode, closeButton: closeButton)
    doordeck.delegate = self
    doordeck.Initialize()
  }
  
  @objc func showUnlock(_ isNfc: Bool, completion: @escaping (Bool) -> Void) {
      DispatchQueue.main.async {
          if let doordeck = self.doordeck {
              doordeck.showUnlockScreen(isNfc ? .nfc : .qr, success: {
                  print("success")
                  completion(true) 
              }) {
                  print("fail")
                  completion(false)
              }
          }
      }
  }

  
}

extension RNDoordeckSdk: DoordeckProtocol {
  func authenticated() {
    
  }
  
  func verificationNeeded() {

  }

  func newAuthTokenRequired() -> AuthTokenClass {
      return AuthTokenClass("")
  }

  func unlockSuccessful() {

  }

}

